bus-mapping/src/evm/opcodes/mload.rs

```diff
@@ -97,81 +97,55 @@ impl Opcode for Mload {
 mod mload_tests {
     use super::*;
     use crate::{
+        bytecode,
         evm::{
-            EvmWord, GasCost, GasInfo, Memory, OpcodeId, ProgramCounter, Stack,
+            EvmWord, GasCost, GasInfo, OpcodeId, ProgramCounter, Stack,
             StackAddress,
         },
-        BlockConstants, ExecutionTrace,
+        external_tracer, BlockConstants, ExecutionTrace,
     };
     use pasta_curves::pallas::Scalar;
 
     #[test]
     fn mload_opcode_impl() -> Result<(), Error> {
-        let trace = r#"
-        [
-            {
-            "pc": 7,
-            "op": "MLOAD",
-            "gas": 79,
-            "gasCost": 3,
-            "depth": 1,
-            "stack": [
-                "40"
-            ],
-            "memory": [
-                "0000000000000000000000000000000000000000000000000000000000000000",
-                "0000000000000000000000000000000000000000000000000000000000000000",
-                "0000000000000000000000000000000000000000000000000000000000000080"
-            ]
-            },
-              {
-                "pc": 8,
-                "op": "STOP",
-                "gas": 76,
-                "gasCost": 0,
-                "depth": 1,
-                "stack": [
-                  "80"
-                ],
-                "memory": [
-                  "0000000000000000000000000000000000000000000000000000000000000000",
-                  "0000000000000000000000000000000000000000000000000000000000000000",
-                  "0000000000000000000000000000000000000000000000000000000000000080"
-                ]
-              }
-        ]
-        "#;
+        let code = bytecode! {
```

Comment:
Here's an idea I would like to discuss.  I think that adding the trace generation into the opcode unit tests has 2 negative sides:
1. Running the tests now requires having go installed and building go packages.
2. While inspecting the unit tests, now it's not obvious what's the trace, because it's being generated in runtime.  So debugging / development can becomes a bit more opaque.

Nevertheless I think having an integrated tool to generate the traces is great, because otherwise it's a very manual process and everyone that needs to add an opcode needs to figure out a way to generate traces, which is not straight forward without having a tool.  So here's my idea:
- have a test folder for evm traces vector tests
- have a rust binary that generates all the evm traces vector tests.  The code for this binary can be extended when new opcodes are added in the bus-mapping.
- In the opcodes unit tests, read the traces vector tests to use as input.
So my proposal is to separate the testing from the test vector generation, and join them by having json files (the traces) in a test vector folder that are created with the test vector generation binary, and are consumed by the opcodes unit tests.  This way it's easy to inspect the trace that a unit file is using, and the generation of the test vector traces don't need to happen during unit testing (eliminating the dependency to go for the general case).  In this proposal, if someone adds code to generate new traces, that person would also commit the new test vectors.  We could even have a github action that generates the test vectors just to verify that the code and the test vectors match.